src/ucs/sys/sock.h

```diff
@@ -105,6 +108,71 @@ ucs_status_t ucs_socket_connect(int fd, const struct sockaddr *dest_addr);
  */
 ucs_status_t ucs_socket_connect_nb_get_status(int fd);
 
+/**
+ * Non-blocking send a data on the connected (or bound connectionless)
+ * socket referred to by the file descriptor `fd`.
+ *
+ * @param [in]      fd          Socket fd.
+ * @param [in]      data        A pointer to a buffer containing the data to
+ *                              be transmitted.
+ * @param [in/out]  length_p    The length, in bytes, of the data in buffer
+ *                              pointed to by the buf parameter. The amount of
+ *                              data transmitted is written to this argument.
+ *
+ * @return UCS_OK on success, UCS_ERR_CANCELED if connection closed,
+ *         UCS_ERR_IO_ERROR on failure.
+ */
+ucs_status_t ucs_socket_send_nb(int fd, const void *data, size_t *length_p);
+
+
+/**
+ * Non-blocking receive a data from the connected (or bound connectionless)
+ * socket referred to by the file descriptor `fd`.
+ *
+ * @param [in]      fd          Socket fd.
+ * @param [in]      data        A pointer to a buffer to receive the incoming
+ *                              data.
+ * @param [in/out]  length_p    The length, in bytes, of the data in buffer
+ *                              pointed to by the buf parameter. The amount of
```

Comment:
what is buf parameter?
the same is relevant for other calls

Reply:
fixed